core/conversion/evaluators/prim.cpp

```diff
@@ -101,6 +102,34 @@ auto prim_registrations =
                         return c10::optional<torch::jit::IValue>(std::move(torch::jit::IValue(list)));
                       }
                     }})
+        .evaluator({c10::Symbol::fromQualString("prim::dtype"),
+                    [](const torch::jit::Node* n, kwargs& args) -> c10::optional<torch::jit::IValue> {
+                      auto input = args.at(n->input(0));
+                      if (input.isITensor()) {
+                        auto trt_dtype = input.ITensor()->getType();
```

Comment:
We have a function TRTDataTypeToScalarType, as well as an optional version which can gracefully tell you there is no type

Reply:
done